src/lib.rs

```diff
@@ -126,15 +126,49 @@ pub trait Renderable: Send + Sync {
     fn render(&self, context: &mut Context) -> Result<Option<String>>;
 }
 
+pub type TemplateLocation = Path;
+pub trait FileRepository {
+    fn read_template_file(&self, path: &TemplateLocation) -> Result<String>;
+}
+
+/// FileSystem to load files relative to the root
+pub struct LocalFileRepository {
+    root: PathBuf,
+}
+
+impl FileRepository for LocalFileRepository {
+    fn read_template_file(&self, relative_path: &TemplateLocation) -> Result<String> {
+        let path = self.root.clone().join(relative_path);
+
+        if !path.exists() {
+            return Err(Error::from(&*format!("{:?} does not exist", path)));
+        }
+        let mut file = try!(File::open(path));
+
+        let mut content = String::new();
+        file.read_to_string(&mut content)?;
+        Ok(content)
+    }
+}
+
 /// Options that `liquid::parse` takes
-#[derive(Default)]
 pub struct LiquidOptions {
     /// Holds all custom block-size tags
     pub blocks: HashMap<String, Box<Block>>,
     /// Holds all custom tags
     pub tags: HashMap<String, Box<Tag>>,
     /// The path to which paths in include tags should be relative to
-    pub file_system: Option<PathBuf>,
+    pub file_repository: Box<FileRepository>,
```

Comment:
Yeah, for now, I feel a `Box` is better.  As I'm learning the code, I'm considering how I want to refactor it.  Maybe after the refactoring, what should be done will become more clear.

My idea is to turn LiquidOptions into a Parser (a Template Builder).  I plan to move filters into the Parser and exposing clients to set their own filters.  Not sure what else will fall out of these changes.